xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/ConfigurableClass.xml

```diff
@@ -194,9 +201,17 @@ $doc.dropPermissions()##
         {{error}}Internal error: All objects were filtered out for application: $appName.{{/error}}
 
       #else
-        #set($formAction = $xwiki.getURL($app.getFullName(), 'save'))
-        #set($formId = "${section.toLowerCase()}_${app.getFullName()}")
-        #set($escapedAppName = $escapetool.xml($app.getFullName()))
+        #if ($editor == "globaladmin")
+          #set($escapedAppName = $escapetool.xml($app.getFullName()))
+          #set($formAction = $xwiki.getURL($app.getFullName(), 'save'))
+          #set($formId = "${section.toLowerCase()}_${app.getFullName()}")
+        #else
+          ## when the application is configured at the space level, it uses the same form but with some different parameters
+          ## escapedAppName, formAction and formId are changed in order to be relative to the space document configuration (WebPreferences)
+          #set($escapedAppName = $escapetool.xml($doc.getFullName()))
+          #set($formAction = $xwiki.getURL($doc.getFullName(), 'save'))
+          #set($formId = "${section.toLowerCase()}_${doc.getFullName()}")
+        #end
```

Comment:
You should refactor this since the the code inside the "if" and "else" is the same except one is using $app and the other $doc. You should introduce a local macro for that.